IRescue/UserLocalisation/Particle/Algos/Resamplers/MultinomialResampler.cs

```diff
@@ -0,0 +1,84 @@
+﻿// <copyright file="MultinomialResampler.cs" company="Delft University of Technology">
+// Copyright (c) Delft University of Technology. All rights reserved.
+// </copyright>
+
+namespace IRescue.UserLocalisation.Particle.Algos.Resamplers
+{
+    using System;
+    using MathNet.Numerics.LinearAlgebra;
+
+    /// <summary>
+    /// Resamples particles using the multinomial algorithm.
+    /// </summary>
+    public class MultinomialResampler : IResampler
+    {
+        /// <summary>
+        /// Resamples Particles using a Multinomial algorithm
+        /// </summary>
+        /// <param name="particles">The Particles to resample</param>
+        /// <param name="weights">The Weights of the Particles</param>
+        public void Resample(Matrix<float> particles, Matrix<float> weights)
+        {
+            int j = 0;
+            foreach (Vector<float> column in weights.EnumerateColumns())
+            {
+                int[] indexes = Multinomial(column);
+                float[] newparticles = new float[particles.RowCount];
+                float[] newweights = new float[particles.RowCount];
+
+                for (int i = 0; i < particles.RowCount; i++)
+                {
+                    newparticles[i] = particles[indexes[i], j];
+                    newweights[i] = weights[indexes[i], j];
+                }
+
+                particles.SetColumn(j, newparticles);
+                weights.SetColumn(j, newweights);
+                j++;
+            }
+        }
+
+        /// <summary>
+        /// Selects a list of indexes of the Particles that will survive the resampling.
+        /// </summary>
+        /// <param name="weights">The Weights of the Particles in a certain dimension</param>
+        /// <returns>The list with indexes of the Particles that are chosen by the resample algorithm</returns>
+        private static int[] Multinomial(Vector<float> weights)
+        {
+            int[] listout = new int[weights.Count];
+            CumSum(weights);
+            Random random = new Random();
```

Comment:
Why create a new Random each time that this function is called? Isn't this unnecessary and extra object creation time. Secondly, Random objects (according to documentation) created within 15ms of each other output the same values due to the way the seed is provided, this would remove part of the randomness.